### PR TITLE
System.currentTimeMillis() 🤮 Instant & Duration 😎

### PR DESCRIPTION
**Pull request by P3ridot**

### CODE REVIEW

File: plugin/src/main/java/net/dzikoysk/funnyguilds/config/ConfigurationFactory.java [hunk 1]
```diff
@@ -45,6 +46,7 @@ public PluginConfiguration createPluginConfiguration(File pluginConfigurationFil
         return ConfigManager.create(PluginConfiguration.class, (it) -> {
             it.withConfigurer(new OkaeriValidator(new YamlBukkitConfigurer(), true), new SerdesCommons());
             it.withSerdesPack(registry -> {
+                registry.register(new DurationTransformer());
```
Comment (PinkLolicorn) [inline plugin/src/main/java/net/dzikoysk/funnyguilds/config/ConfigurationFactory.java:49]: Redundant. See: `new SerdesCommons()` in line 47.

File: plugin/src/main/java/net/dzikoysk/funnyguilds/config/PluginConfiguration.java [hunk 2]
```diff
@@ -1125,15 +1121,13 @@ public void loadProcessedProperties() {
 
         this.tntProtection.time.passingMidnight = this.tntProtection.time.startTime.getTime().isAfter(this.tntProtection.time.endTime.getTime());
 
-        if (!"v1_8_R1".equals(Reflections.SERVER_VERSION) && !"v1_8_R3".equals(Reflections.SERVER_VERSION)) {
+        if (!"v1_8_R3".equals(Reflections.SERVER_VERSION)) {
```
Comment (PinkLolicorn) [inline plugin/src/main/java/net/dzikoysk/funnyguilds/config/PluginConfiguration.java:1124]: Seems unrelated to the main purpose of the PR?

File: plugin/src/main/java/net/dzikoysk/funnyguilds/listener/PlayerDeath.java [hunk 3]
```diff
@@ -103,11 +104,11 @@ public void onDeath(PlayerDeathEvent event) {
...
 
         if (config.rankFarmingProtect) {
-            Long attackTimestamp = attackerCache.wasAttackerOf(victim);
-            Long victimTimestamp = attackerCache.wasVictimOf(attacker);
+            Instant attackTimestamp = attackerCache.wasAttackerOf(victim);
+            Instant victimTimestamp = attackerCache.wasVictimOf(attacker);
 
             if (attackTimestamp != null) {
-                if (attackTimestamp + (config.rankFarmingCooldown * 1000L) >= System.currentTimeMillis()) {
+                if (attackTimestamp.plusSeconds(config.rankFarmingCooldown).compareTo(Instant.now()) >= 0) {
```
Comment (PinkLolicorn) [inline plugin/src/main/java/net/dzikoysk/funnyguilds/listener/PlayerDeath.java:111]: Seems like rankFarmingCooldown should be converted to Duration too!

**Review by PinkLolicorn — CHANGES_REQUESTED** [review]

File: plugin/src/main/java/net/dzikoysk/funnyguilds/config/PluginConfiguration.java (hunk 2 above)
Comment (P3ridot) [reply plugin/src/main/java/net/dzikoysk/funnyguilds/config/PluginConfiguration.java:1124]: It's not only this PR has unrelated changes so don't mind about that. Here it's normal

Comment (P3ridot) [reply plugin/src/main/java/net/dzikoysk/funnyguilds/config/PluginConfiguration.java:1129]: Yes

**Review by dzikoysk — APPROVED** [review]
